pkg/gadgets/trace/sni/tracer/tracer.go

```diff
@@ -92,17 +97,41 @@ func parseSNIEvent(sample []byte, netns uint64) (*types.Event, error) {
 }
 
 // --- Registry changes
+// TODO: This can be optimized a lot after using NewInstance() for everything
 
 func (g *GadgetDesc) NewInstance() (gadgets.Gadget, error) {
 	return &Tracer{}, nil
 }
 
 func (t *Tracer) Init(gadgetCtx gadgets.GadgetContext) error {
-	// TODO: Clean up
+	if err := t.install(); err != nil {
+		t.Close()
+		return fmt.Errorf("installing tracer: %w", err)
+	}
+
+	t.ctx, t.cancel = gadgetcontext.WithTimeoutOrCancel(gadgetCtx.Context(), gadgetCtx.Timeout())
+	return nil
+}
+
+func (t *Tracer) install() error {
+	// TODO: It is ugly. Clean it up.
```

Comment:
I tried to handle this on https://github.com/inspektor-gadget/inspektor-gadget/commit/c4e1a45a1fae98d5e5e124046de2c28cdcacc235. Feel free to take and squash it if you agree.